plugins/tokens/create.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const urlLib = require('url');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/tokens',
+    config: {
+        description: 'Create a new token',
+        notes: 'Create a specific token',
+        tags: ['api', 'tokens'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const tokenFactory = request.server.app.tokenFactory;
+            const userFactory = request.server.app.userFactory;
+            const username = request.auth.credentials.username;
+            let user;
+
+            return userFactory.get({ username })
+            .then((theUser) => {
```

Comment:
Hmm why not just leave this as `.then((user) => ...`, and have the `.then(tokens)` inside this user scope? I think that would be cleaner.